clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ImageCachingAgent.groovy

```diff
@@ -102,10 +102,10 @@ class ImageCachingAgent implements CachingAgent, AccountAware, DriftMetric {
     Collection<CacheData> namedImageCacheData = new ArrayList<>(images.size())
 
     for (Image image : images) {
-      if (includePublicImages || !image.public) {
+      if (includePublicImages || !image.public || image.ownerId == account.accountId) {
```

Comment:
Might need some love in the case where `clouddriver` isn't indexing the owning account.